pkgs/development/python-modules/bsddb3/default.nix

```
@@ -1,14 +1,14 @@
 {
   lib,
-  buildPythonPackage,
+  python3Packages,
+  pythonAtLeast,
+  python,
   fetchPypi,
   setuptools,
   pkgs,
-  python,
-  pythonOlder,
 }:
 
-buildPythonPackage rec {
+python3Packages.buildPythonPackage rec {
```

```suggestion
buildPythonPackage rec {
```
*(And also do the relevant change in the top attrset)*

`python3Packages` only ever has to be used when dealing with packages outside of the `python-modules` directory. Inside the directory, the top attrset's values are also populated by `python3Packages`'s values.

*Also, even if you wanted to do this, you'd have to use `pythonPackages`, since, it turns out `python3Packages` is always an alias to `python311Packages`.*

---

Thanks for the explanation, I will keep this in mind for future PRs.

---

One check seems to fail after having removed the `python3Packages` : https://github.com/NixOS/nixpkgs/actions/runs/9528874875/job/26267015309?pr=311198

---

I have no idea why it is failing only for `bsddb3`, for `berkeleydb`, it does work as you described